### PR TITLE
Write "companyName" field in JucerHeader.h like Projucer does

### LINKED ISSUE #680: Company name missing from JuceHeader
Currently the ProjectInfo::companyName field doesn't get added from the Jucer file, when the JuceHeader is generated through FRUT.

### PR DESCRIPTION
**Pull request by McMartin**
Resolves #680.

@MartyLake: please review, thanks!

### CODE REVIEW

**Comment by benediktsailer** [comment]
I think there's a small type-o in `cmake/data/JuceHeader.h.in`

It says
`const char* const projectName = "@JUCER_PROJECT_NAME@";@company_name_field@`

on line 26 where it should say 
`
const char* const projectName = "@JUCER_PROJECT_NAME@"; 
const char* const companyName = "@company_name_field@";
`
 instead

**Comment by McMartin** [comment]
@benediktsailer see the changes in `cmake/Reprojucer.cmake` for what `@company_name_field@` will be replaced by. It's more than just the company name. Maybe I should rename `company_name_field` to `company_name_line`.

**Comment by benediktsailer** [comment]
I see! My bad! :)

**Comment by McMartin** [comment]
@benediktsailer No problem. Please do test and let me know whether it works as expected. Thanks!

**Comment by benediktsailer** [comment]
Working like a charm! Thank you!

**Comment by MartyLake** [comment]
If you want to strictly follow the change of behavior, then this PR LGTM as-is.

However, versioning the API change like that is hard to follow if one is not in this exact commit frame but only reviewing one file at a time (in it's text editor for example). Since companyName is not a breaking API change, maybe only adding is less complex and works equally well ? (untested)

```diff
#if ! JUCE_DONT_DECLARE_PROJECTINFO
namespace ProjectInfo
{
    const char* const  projectName    = "@JUCER_PROJECT_NAME@";
+   const char* const  companyName    = "@JUCER_COMPANY_NAME@";
    const char* const  versionString  = "@JUCER_PROJECT_VERSION@";
    const int          versionNumber  = @JUCER_PROJECT_VERSION_AS_HEX@;
}
```

**Comment by McMartin** [comment]
Always having `ProjectInfo::companyName` should indeed work, but `Reprojucer.cmake` tries to reproduce the Projucer behavior as closely as possible. Making the code more complex to achieve that is fine.
